doc/cephfs/multimds.rst

```diff
@@ -311,3 +313,151 @@ the ``bal_rank_mask`` should be set to ``0x0``. For example:
+without affecting the performance of other concurrent workloads.
+
+
+This option can be set via:
+
+::
+
+    setfattr -n ceph.dir.bal.mask -v 1,2 /usr/share/images
+    setfattr -n ceph.dir.bal.mask -v 3,4 /usr/share/backups
+
+.. note:: The balance_automate setting should be `true` for this option to be turned on.
+
+``/usr/share/images`` and ``/usr/share/backups`` are distributed within mds rank 1~2 and 3~4, respectively.
+
+Similar to ``ceph.dir.pin``, the ``ceph.dir.bal.mask`` is also inherited from its closest parent.
+This manner involves configuring the ``ceph.dir.bal.mask`` for a directory, which consequently
+impacts all its descendants. However, the parent ``ceph.dir.bal.mask`` can be overridden
+by setting the child directory’s value. For example:
+
+::
+
+    mkdir -p /usr/share/images/jpg
+    setfattr -n ceph.dir.bal.mask -v 1,2 /usr/share/images
+    # /usr/share/images and /usr/share/images/jpg are now freely moved among mds ranks 1 and 2
+    setfattr -n ceph.dir.bal.mask -v 3,4 /usr/share/images/jpg
+    # /usr/share/images/jpg now moved among mds ranks 3 and 4
+
+The option can be unset via:
+
+::
+
+    setfattr -x ceph.dir.bal.mask /usr/share/images/jpg
+
+The value of the ``ceph.dir.bal.mask`` for  ``/usr/share/images/jpg`` is unset and replaced with their inherited value from
+its nearest parent. If the ``ceph.dir.bal.mask`` for ``/usr/share/images`` is configured with
+a valid value, ``/usr/share/images/jpg`` will be distributed according to its parent’s settings.
+
+This can override the file system setting ``bal_rank_mask``. For example:
+
+::
+
+    cephfs fs set cephfs bal_rank_mask 0xf
+
+
+Initially, the balancer dynamically partitions the file system within MDS rank
+0 to 3 because the ``bal_rank_mask`` is set to ``0xf``.
+
+If the ``ceph.dir.bal.mask`` for the root ``/`` directory is set to ``0,1,2,3,4,5``,
+the ``bal_rank_mask`` will be overridden. In this way, the balancer dynamically
+distributes the unpinned subtrees of the root ``/`` directory from ``0`` to ``5`` mds ranks.
+
+.. note:: Users may prefer to use the file system setting `bal_rank_mask` for simplicity.
+
+::
+
+    setfattr -n ceph.dir.bal.mask -v 0,1,2,3,4,5 /
+
+``ceph.dir.bal.mask`` overrides parent ``ceph.dir.pin`` and vice versa. For example:
+
+::
+
+    mkdir -p /usr/share/images/jpg
+    setfattr -n ceph.dir.pin -v 1 /usr/share/images
+    setfattr -n ceph.dir.bal.mask -v 2,3 /usr/share/images/jpg
+
+The ``/usr/share/images`` directory will be pinned to rank 1, while the ``/usr/share/images/jpg``
+directory may be dynamically split into rank 2 and 3.
+
+While `ceph.dir.bal.mask` functions similarly to `ceph.dir.pin` when it has only one mds rank,
+its distinct advantage lies in its flexibility.
+Unlike `ceph.dir.pin`, `ceph.dir.bal.mask` can be expanded to multiple mds ranks or reduced to fewer mds ranks.
+Therefore, in scenarios requiring expansion to multiple mds ranks,
+it is recommended to employ `ceph.dir.bal.mask`.
```

Comment:
On the other hand, if the admin wants the automated balancer to be off, then `ceph.dir.pin` or ephemeral pinning is the way to distribute load.